pkg/controllers/edgeconnect/controller.go

```diff
@@ -319,6 +324,11 @@ func (controller *Controller) reconcileEdgeConnectRegular(ctx context.Context, e
 
 	_log := log.WithValues("namespace", edgeConnect.Namespace, "name", edgeConnect.Name, "deploymentName", desiredDeployment.Name)
 
+	err := controller.hostAlias(ctx, desiredDeployment, edgeConnect.Name, edgeConnect.Namespace)
+	if err != nil {
+		return err
+	}
+
 	if err := controllerutil.SetControllerReference(edgeConnect, desiredDeployment, controller.scheme); err != nil {
 		return errors.WithStack(err)
 	}
```

Comment:
Couldn't the `createOrUpdateEdgeConnectDeployment` function also be used here?

`createOrUpdateEdgeConnectDeployment` is only used by `reconcileEdgeConnectProvisioner` currently, but the naming of the function and the contents of it seems generic enough so we might be able to use it here aswell.

Reply:
good point!

Reply:
@0sewa0 wdyt if we merge as is, and I'll create a new ticket to refactor it separately?